src/test/java/org/assertj/core/api/throwable/ThrowableAssert_hasCauseSameAs_Test.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.api.throwable;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.ThrowableAssert;
+import org.assertj.core.api.ThrowableAssertBaseTest;
+import org.junit.jupiter.api.Test;
+
+public class ThrowableAssert_hasCauseSameAs_Test extends ThrowableAssertBaseTest {
+
+  private Throwable npe = new NullPointerException();
+
+  @Override
+  protected ThrowableAssert invoke_api_method() {
+    return assertions.hasCauseSameAs(npe);
+  }
+
+  @Override
+  protected void verify_internal_effects() {
+    verify(throwables).assertHasCauseSameAs(getInfo(assertions), getActual(assertions), npe);
+  }
+
+  @Test
+  public void should_fail_if_actual_and_expected_cause_are_not_same_instance() {
```

Comment:
remove test as we check we delegate to `assertHasCauseSameAs` and this is tested separately